x-pack/plugins/timelines/public/components/actions/timeline/cases/add_to_case_action.tsx

```diff
@@ -0,0 +1,317 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { isEmpty } from 'lodash';
+import React, { memo, useState, useCallback, useMemo } from 'react';
+import {
+  EuiPopover,
+  EuiButtonIcon,
+  EuiContextMenuPanel,
+  EuiText,
+  EuiContextMenuItem,
+  EuiToolTip,
+} from '@elastic/eui';
+
+import { Case, CaseStatuses, StatusAll } from '../../../../../../cases/common';
+import { Ecs } from '../../../../../common/ecs';
+import { useGetUserCasesPermissions } from '../../../../hooks/use_get_user_cases_permissions';
+import { useKibana } from '../../../../../../../../src/plugins/kibana_react/public';
+import { TimelinesStartServices } from '../../../../types';
+import { ActionIconItem } from '../../action_icon_item';
+import { CreateCaseFlyout } from './create/flyout';
+import { createUpdateSuccessToaster } from './helpers';
+import * as i18n from './translations';
+
+export interface AddToCaseActionProps {
+  ariaLabel?: string;
+  ecsRowData: Ecs;
+  appId: string;
+}
+// FIXME: DEDUPE
```

Comment:
👍

Reply:
do not forget about it

Reply:
let's pass the appId and deepLinkId as a props so we can remove the de-dupe and it will also allow o11y to use it